apps/studio/src/server/modules/page/__tests__/page.router.test.ts

```diff
@@ -0,0 +1,1206 @@
+    })
+
+    it("should throw 401 if not logged in", async () => {
+      const unauthedSession = applySession()
+      const unauthedCaller = createCaller(createMockRequest(unauthedSession))
+      const pageUpdateArgs = createPageUpdateArgs(pageToUpdate)
+      const result = unauthedCaller.updatePageBlob(pageUpdateArgs)
+
+      await expect(result).rejects.toThrowError(
+        new TRPCError({ code: "UNAUTHORIZED" }),
+      )
+    })
+
+    it("should return 404 if page does not exist", async () => {
+      // Arrange
+      const pageUpdateArgs = createPageUpdateArgs(pageToUpdate)
+
+      // Act
+      const result = caller.updatePageBlob({
+        ...pageUpdateArgs,
+        pageId: 999999, // should not exist
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError(
+        new TRPCError({
+          code: "NOT_FOUND",
+          message: "Resource not found",
+        }),
+      )
+    })
+
+    it("should return 422 if content is not valid", async () => {
+      // Arrange
+      const pageUpdateArgs = createPageUpdateArgs(pageToUpdate)
+
+      // Act
+      const result = caller.updatePageBlob({
+        ...pageUpdateArgs,
+        content: "do not match the shape",
+      })
+
+      // Assert
+      await expect(result).rejects.toThrowError("Invalid page content")
+    })
+
+    it("should update draft page blob if args are valid and has current draft", async () => {
+      // Arrange
+      const pageUpdateArgs = createPageUpdateArgs(pageToUpdate)
+
+      // Act
+      const result = await caller.updatePageBlob(pageUpdateArgs)
+
+      // Assert
+      const actual = await db
+        .selectFrom("Blob")
+        .where("id", "=", pageToUpdate.draftBlobId)
+        .select("content")
+        .executeTakeFirstOrThrow()
+      expect(actual.content).toEqual(result.content)
+    })
+
+    it("should create draft page blob if args are valid and without current draft", async () => {
+      // Arrange
+      const { page: publishedPageToUpdate } = await setupPageResource({
+        resourceType: "Page",
+        state: "Published",
+        userId: session.userId,
+      })
+      expect(publishedPageToUpdate.publishedVersionId).not.toBeNull()
+      expect(publishedPageToUpdate.draftBlobId).toBeNull()
+      const pageUpdateArgs = createPageUpdateArgs(publishedPageToUpdate)
+
+      // Act
+      const result = await caller.updatePageBlob(pageUpdateArgs)
+
+      // Assert
+      const actual = await db
+        .selectFrom("Blob")
+        .innerJoin("Resource", "Resource.draftBlobId", "Blob.id")
+        .where("Resource.id", "=", publishedPageToUpdate.id)
+        .select([
+          "Blob.content",
+          "Resource.publishedVersionId",
+          "Resource.draftBlobId",
+        ])
+        .executeTakeFirstOrThrow()
+      expect(actual).toMatchObject({
+        content: result.content,
+        publishedVersionId: publishedPageToUpdate.publishedVersionId,
+        draftBlobId: expect.any(String),
```

Comment:
leave it to you but shuold we check if this id is different from the published blob id? this is to check that we don't override already published blobs